vendor/k8s.io/kubernetes/pkg/kubectl/cmd/drain.go

```diff
@@ -188,15 +193,24 @@ func NewCmdDrain(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
 	cmd.Flags().BoolVar(&options.DeleteLocalData, "delete-local-data", false, "Continue even if there are pods using emptyDir (local data that will be deleted when the node is drained).")
 	cmd.Flags().IntVar(&options.GracePeriodSeconds, "grace-period", -1, "Period of time in seconds given to each pod to terminate gracefully. If negative, the default value specified in the pod will be used.")
 	cmd.Flags().DurationVar(&options.Timeout, "timeout", 0, "The length of time to wait before giving up, zero means infinite")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on")
 	return cmd
 }
 
 // SetupDrain populates some fields from the factory, grabs command line
 // arguments and looks up the node using Builder
 func (o *DrainOptions) SetupDrain(cmd *cobra.Command, args []string) error {
+	o.Selector = cmdutil.GetFlagString(cmd, "selector")
+
 	var err error
-	if len(args) != 1 {
```

Comment:
With this new if you're swallowing the case where nothing was passed and you should return error. I think the best option would be to:

1. check args == 0 -> error out
2. check if args and selector is passed -> error out